src/sentry/tasks/check_monitors.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import absolute_import, print_function
+
+import logging
+
+from django.utils import timezone
+
+from sentry.models import Monitor, MonitorStatus, MonitorType
+from sentry.tasks.base import instrumented_task
+
+
+logger = logging.getLogger('sentry')
+
+
+@instrumented_task(name='sentry.tasks.check_monitors', time_limit=15, soft_time_limit=10)
+def check_monitors():
+    qs = Monitor.objects.filter(
```

Comment:
Definitely need an index for this query. Probably fully on `(type, status, next_checkin)`. Can get away without having `status` in there if we think that case is pretty rare.

Reply:
I think we can ignore status and just NULL the field if you delete/disable the monitor.